api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditService.kt

```diff
@@ -3,9 +3,18 @@ package org.pmiops.workbench.actionaudit
 import java.util.logging.Logger
 
 interface ActionAuditService {
-    fun send(event: ActionAuditEvent)
+    fun send(event: ActionAuditEvent) {
+        send(setOf(event))
+    }
 
     fun send(events: Collection<ActionAuditEvent>)
 
+    /**
+     * Sends a collection of events plus an event with ActionType.COMMENT and the given comment
+     * string as the newValueMaybe. The COMMENT action derives its other properties from the
+     * first event from the original collection.
```

Comment:
I see where you're going, and how something like this could be useful, but there are a couple of things that are of some concern:

1. The model doesn't really support multiple action types per actionId.
2. A utility like this would naturally belong to the `AuditActionEvent.Builder` class, not this service.
3. It's ambiguous to me when this would be used. If we're capturing administrator comments, then that makes sense to me as a dedicated `TargetProperty` that can be added to whatever the administrator (agent) is doing (action). So from that standpoint, we don't need a global comment.
4. If it's for developers to add arbitrary strings, then things can start to get messy in a hurry. It will be tempting to just stuff things into a comment when you don't know where they would logically go in the schema. That's better than not having the information, but one of my original requirements was for a high degree of relational query-ability.
5. If we want to support stand-alone comment actions (independent of other events), then that's another action type (as you have here) with a target of whatever context the admin is looking at (user, workspace, etc).
6. If we decide to migrate some date that's in comments into a more structured form, that's really tough to do compared to more structured data.
7. It's not stable to take the first element from a `Collection`, since they're not ordered.

Reply:
Chatted offline – check out the latest commit for a different approach to this.